azure/notebooks/azure_mnmg_xgboost.ipynb

```diff
@@ -0,0 +1,1287 @@
+    "ManagedImageLocation: <some chosen location>\n",
+    "```\n",
+    "\n",
+    "##### Make note of the `ManagedImageId`. This is the custom image id we will use. \n",
+    "As shown above the `ManagedImageId` will look something like : `/subscriptions/12345/resourceGroups/myown-rg/providers/Microsoft.Compute/images/myCustomImage`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### f. Set up customized VM information and clear default dask config. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once you have the custom VM id, you should reset the default VM image information in `dask.config`. The default image value loaded in `dask.config` is that of a basic Ubuntu Server 18.04 LTS (the one that you already customized). You wouldn't want to use a vanilla VM image after all such trouble, would you?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ManagedImageId = <value from the prompt> # or the customized VM id if you already have resource id of the customized VM from a previous run. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dask.config.set({\"cloudprovider.azure.azurevm.vm_image\":{}})\n",
+    "config = dask.config.get(\"cloudprovider.azure.azurevm\", {})\n",
+    "print(config)\n",
+    "vm_image = {\"id\": ManagedImageId}\n",
+    "print(vm_image)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### <span style=\"color:#8735fb; font-size:20pt\"> Step 1.2: Start the VM Cluster in Azure. </span>\n",
+    "\n",
+    "Here, if you have used Option 1, i.e., the NVIDIA VM image, pass blank string for `vm_image` information. \n",
+    "\n",
+    "For Option 2, pass the `vm_image` information that you got from the output of `packer` run as a parameter to `AzureVMCluster`. \n",
+    "\n",
+    "Also turn off the bootstrapping of the VM image by passing `bootstrap=False`. This will turn off installation of the dependencies in the VM, since we already have them on our custom VM image in either cases."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**NOTE:** The rest of the notebook should be the same irrespective of whether you chose Option 1 or Option 2.\n",
+    "\n",
+    "**NOTE on $n\\_workers$:** The number of actual workers that our cluster would have is not always equal to the number of VMs spawned i.e. the value of $n\\_workers$ passed in. If the number of GPUs in the chosen `vm_size` is $G$ and number of VMs spawned is $n\\_workers$, then we have then number of actual workers $W = n\\_workers \\times G$. For example, for Standard_NC12s_v3 VMs that have 2 V100 GPUs per VM, for $n\\_workers=2$, we have $W = 2 \\times 2=4$."
```

Comment:
Missing step to accept EULA for using the image, I use the Azure CLI: https://docs.microsoft.com/en-us/cli/azure/vm/image/terms?view=azure-cli-latest

Reply:
Huh, weird. I did not need to do this step. I think that may be because the image was probably used before in this subscription. I'll add the step here.

Reply:
It also throws me an error :
```bash
>>> az vm image terms accept --urn "nvidia:ngc_azure_17_11:ngc-base-version-21-02-2:21.02.2" --verbose
(AuthorizationFailed) The client '<username>@nvidia.com' with object id '87fd9418-cd7b-42c1-9d80-ea856b6b3a79' does not have authorization to perform action 'Microsoft.MarketplaceOrdering/offerTypes/publishers/offers/plans/agreements/read' over scope '/subscriptions/<subscription ID>' or the scope is invalid. If access was recently granted, please refresh your credentials.
```